objects/src/accounts/storage/slot.rs

```diff
@@ -67,8 +67,8 @@ impl TryFrom<u16> for StorageSlotType {
     type Error = String;
 
     fn try_from(value: u16) -> Result<Self, Self::Error> {
-        let data_type = value as u8;
-        let value_arity = (value >> 8) as u8;
```

Comment:
The bitwise operations are not needed [ref](https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast):

> Casting from a larger integer to a smaller integer (e.g. u32 -> u8) will truncate

Example:

```rust
fn main() {
    let v = 1 + 2u32.pow(9);
    println!("{} {}", v, v as u8);
}
```

will print:

> 513 1

Reply:
ah nice didn't know